tests/test_drvsupport.py

```diff
@@ -225,73 +251,73 @@ def test_no_write_driver_cannot_write(tmpdir, driver, testdata_generator, monkey
+@pytest.mark.parametrize(
+    "driver",
+    [
+        driver
+        for driver, raw in supported_drivers.items()
+        if "w" in raw and "a" not in raw
+    ],
+)
+def test_no_append_driver_cannot_append(
+    tmpdir, driver, testdata_generator, monkeypatch
+):
     """
     Test if a driver that supports write and not append cannot also append
 
     If this test fails, it should be considered to enable append support for the respective driver in drvsupport.py.
 
     """
 
-    monkeypatch.setitem(fiona.drvsupport.supported_drivers, driver, 'raw')
+    monkeypatch.setitem(fiona.drvsupport.supported_drivers, driver, "raw")
 
     if driver == "BNA" and GDALVersion.runtime() < GDALVersion(2, 0):
         pytest.skip("BNA driver segfaults with gdal 1.11")
 
     path = str(tmpdir.join(get_temp_filename(driver)))
-    schema, crs, records1, records2, test_equal, create_kwargs = testdata_generator(driver, range(0, 5), range(5, 10))
+    schema, crs, records1, records2, test_equal, create_kwargs = testdata_generator(
+        driver, range(0, 5), range(5, 10)
+    )
 
     # If driver is not able to write, we cannot test append
-    if (driver in driver_mode_mingdal['w'] and
-            get_gdal_version_num() < calc_gdal_version_num(*driver_mode_mingdal['w'][driver])):
+    if driver in driver_mode_mingdal[
+        "w"
+    ] and get_gdal_version_num() < calc_gdal_version_num(
+        *driver_mode_mingdal["w"][driver]
+    ):
         return
 
     # Create test file to append to
-    with fiona.open(path, 'w',
-                    driver=driver,
-                    crs=crs,
-                    schema=schema,
-                    **create_kwargs) as c:
+    with fiona.open(
+        path, "w", driver=driver, crs=crs, schema=schema, **create_kwargs
+    ) as c:
 
         c.writerecords(records1)
 
-    is_good = True
     try:
-        with fiona.open(path, 'a',
-                        driver=driver) as c:
+        with fiona.open(path, "a", driver=driver) as c:
             c.writerecords(records2)
+    except Exception as exc:
+        log.exception("Caught exception in trying to append.")
 
-        if driver in {'FileGDB', 'OpenFileGDB'}:
-            open_driver = driver
-        else:
-            open_driver = None
-        with fiona.open(path, driver=open_driver) as c:
-            assert c.driver == driver
-            items = list(c)
-            is_good = is_good and len(items) == len(records1) + len(records2)
-            for val_in, val_out in zip(records1 + records2, items):
-                is_good = is_good and test_equal(driver, val_in, val_out)
-    except:
-        is_good = False
+    if driver in {"FileGDB", "OpenFileGDB"}:
+        open_driver = driver
+    else:
+        open_driver = None
 
-    assert not is_good
+    with fiona.open(path, driver=open_driver) as collection:
+        assert collection.driver == driver
+        assert len(collection) == len(records1)
```

Comment:
I think this is the crux of it. If no new records were added and there was no other exception, the format does not support appending.